src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,101 @@
+        {
+            "runtimeFramework",
+            "baseSdkTargetFramework",
+            UpdatedFieldsPropertyKey
+        };
+
+        public static void GetPayloadProperties(IEnumerable<string> disabledFields, IHeartbeatProvider provider)
+        {
+            var enabledProperties = HeartbeatDefaultPayload.RemoveDisabledDefaultFields(disabledFields);
+
+            var payload = new Dictionary<string, HeartbeatPropertyPayload>();
+            foreach (string fieldName in enabledProperties)
+            {
+                try
+                {
+                    switch (fieldName)
+                    {
+                        case "runtimeFramework":
+                            provider.AddHealthProperty(fieldName, HeartbeatDefaultPayload.GetRuntimeFrameworkVer(), true);
+                            break;
+                        case "baseSdkTargetFramework":
+                            provider.AddHealthProperty(fieldName, HeartbeatDefaultPayload.GetBaseSdkTargetFramework(), true);
+                            break;
+                        case UpdatedFieldsPropertyKey:
+                            provider.AddHealthProperty(fieldName, string.Empty, true);
+                            break;
+                        default:
+                            provider.AddHealthProperty(fieldName, "UNDEFINED", false);
+                            break;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    CoreEventSource.Log.FailedToObtainDefaultHeartbeatProperty(fieldName, ex.ToString());
+                }
+            }
+        }
+
+        private static List<string> RemoveDisabledDefaultFields(IEnumerable<string> disabledFields)
+        {
+            List<string> enabledProperties = new List<string>();
+
+            if (disabledFields == null || disabledFields.Count() <= 0)
+            {
+                enabledProperties = DefaultFields.ToList();
+            }
+            else
+            {
+                enabledProperties = new List<string>();
+                foreach (string fieldName in DefaultFields)
+                {
+                    if (!disabledFields.Contains(fieldName, StringComparer.OrdinalIgnoreCase))
+                    {
+                        enabledProperties.Add(fieldName);
+                    }
+                }
+            }
+
+            return enabledProperties;
+        }
+
+        private static string GetBaseSdkTargetFramework()
+        {
+#if NET45
+            return "net45";
+#elif NET46
+            return "net46";
+#elif NETSTANDARD1_3
+            return "netstandard1.3";
+#else
+#error Unrecognized framework
+            return "undefined";
+#endif
+        }
+
+        private static string GetRuntimeFrameworkVer()
+        {
+            // taken from https://github.com/Azure/azure-sdk-for-net/blob/f097add680f37908995fa2fbf6b7c73f11652ec7/src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs#L214
```

Comment:
nit: You do not need this comment any longer. If you really want to keep it - move to the summary doc

Reply:
I doubt it’s useful to keep a reference now that I see others doing similar things. _Removed._